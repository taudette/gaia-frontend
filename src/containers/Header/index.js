import React from 'react';
import { connect } from 'react-redux';
import gaiaLogo from '../../../public/images/header-logo.svg';
import userLogo from '../../../public/images/user.png';
import styles from '../../styles/components/_header.scss';
import { Row, Col } from 'react-bootstrap';

export const HeaderContainer = (props) => (
  <section>
    <Row className={styles.row}>
      <Col xs={6}>
        <div className={styles.logoContainer}>
          <img src={gaiaLogo} alt="Gaia, Inc." />
        </div>
      </Col>
      <Col xs={6}>
        <div className={styles.userContainer}>
          <span><small>{props.user}</small></span>
          <img src={userLogo} alt="user logo" />
        </div>
      </Col>
    </Row>
  </section>
)

const mapStateToProps = (state) => {
  return {
    user: state.user
  }
}

export default connect(mapStateToProps, null)(HeaderContainer)
