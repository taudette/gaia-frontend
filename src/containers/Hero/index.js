import React from 'react';

export const HeroContainer = () => (
  <div>
    Hero
  </div>
)

export default HeroContainer
