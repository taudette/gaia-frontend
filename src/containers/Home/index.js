import React from 'react';
import HeroComponent from '../../components/Hero';
import VideosComponent from '../../components/Videos';
import LoadButtonsComponent from '../../components/LoadButtons';
import { fetchVideos } from '../../actions';
import { connect } from 'react-redux';
import '../../styles/components/_home.scss';

export class HomeContainer extends React.Component {
  constructor(props) {
    super(props)
    //use component state to store videos to be displayed
    this.state = {
      index: 16,
      loading: false,
      listLength: ''
    }
  }

  componentDidMount() {
   this.props.fetchData && this.props.fetchData() || null
  }

  addVideos = (e) => {
    e.preventDefault()
    this.setState({ loading: true })

    //if possible add 16 videos to the screen
    //if not just show the rest remaining
    let endIndex = this.state.index
    const listLength = this.props.videos.length

    if (listLength >= this.state.index + 16) {
      endIndex += 16
    } else {
      endIndex = listLength
    }

    if (this.state.index == listLength) {
      return null
    }

    this.setState({ listLength: listLength })
    this.setState({ index: endIndex })
    this.setState({ loading: false })
  }

  goToTop = (e) => {
    e.preventDefault();
		return window.scrollTo(0,0);
  }

  render() {
    //check if there are as many videos as initially requested and show first batch (16)
    if (this.props.videos) {
      let videoList = this.props.videos.length >= this.state.index ? this.props.videos.slice(0, this.state.index) : this.props.videos
    }
    

    if (this.props.error) {
      return <p>Sorry! Something went wrong</p>
    }

    if (this.props.isFetching) {
      return <p>Loading...</p>
    }

    if (this.props.hasFetched) {
      return (
        <div>
          <HeroComponent hero={this.props.hero} />
          <VideosComponent videos={videoList} loading={this.state.loading} />
          <LoadButtonsComponent addVideos={this.addVideos} goToTop={this.goToTop} endIndex={this.state.index} listLength={this.state.listLength} />
        </div>
      )
    }

    return null
  }
}

const mapStateToProps = (state) => {
  return {
    videos: state.videos,
    hero: state.hero,
    error: state.error,
    isFetching: state.isFetching,
    hasFetched: state.hasFetched
  }
}
const mapDispatchToProps = (dispatch) => {
  return {
    fetchData: () => dispatch(fetchVideos())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(HomeContainer)
