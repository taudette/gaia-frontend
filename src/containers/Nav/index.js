import React from 'react';

export const NavContainer = () => (
  <div>
    Nav Container
  </div>
)

export default NavContainer
