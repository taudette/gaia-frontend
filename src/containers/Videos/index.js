import React from 'react';
import VideosComponent from '../../components/Videos'

export const VideosContainer = () => (
  <div>
    Videos Container
    <VideosComponent />
  </div>
)

export default VideosContainer
