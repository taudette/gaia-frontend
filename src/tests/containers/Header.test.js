import React from 'react';
import { configure, shallow } from 'enzyme';
import { HeaderContainer } from '../../containers/Header';
import ConnectedHeader from '../../containers/Header';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

test('HeaderContainer Container is connected', () => {
  const mockStore = configureStore([]);
  const store = mockStore({});
  const wrapper = shallow(<Provider store={store}>
    <ConnectedHeader />
  </Provider>);
  expect(wrapper.text()).toBe('<Connect(HeaderContainer) />');
})

test('should render Header correctly', () => {
  const wrapper = shallow(<HeaderContainer />);
  expect(wrapper).toMatchSnapshot();
});

