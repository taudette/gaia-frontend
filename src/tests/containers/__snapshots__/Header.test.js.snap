// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <Row
    bsClass="row"
    className="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      xs={6}
    >
      <div
        className="logoContainer"
      >
        <img
          alt="Gaia, Inc."
          src="test-file-stub"
        />
      </div>
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      xs={6}
    >
      <div
        className="userContainer"
      >
        <span>
          <small />
        </span>
        <img
          alt="user logo"
          src="test-file-stub"
        />
      </div>
    </Col>
  </Row>
</header>
`;
